testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -581,3 +584,137 @@ def test_segmentgroup_repr(self):
+        assert_(not c.issuperset(a))
+        assert_(a.issuperset(d))
+        assert_(not d.issuperset(a))
+
+    @staticmethod
+    def _test_concatenate(a, b, c, d):
+        cat_ab = a.concatenate(b)
+        assert_(cat_ab[:len(a)] == a)
+        assert_(cat_ab[len(a):] == b)
+
+        cat_ba = b.concatenate(a)
+        assert_(cat_ba[:len(b)] == b)
+        assert_(cat_ba[len(b):] == a)
+
+        cat_aa = a.concatenate(a)
+        assert_(cat_aa[:len(a)] == a)
+        assert_(cat_aa[len(a):] == a)
+
+        cat_ad = a.concatenate(d)
+        assert_(cat_ad == a)
+
+        cat_da = d.concatenate(a)
+        assert_(cat_da == a)
+
+    @staticmethod
+    def _test_union(a, b, c, d):
+        union_ab = a.union(b)
+        assert_(union_ab.ix.tolist() == sorted(union_ab.ix.tolist()))
+        assert_(list(sorted(set(union_ab.ix))) == list(sorted(union_ab.ix)))
+
+        assert_(a.union(b) == b.union(a))
+        assert_(a.union(a) == a)
+        assert_(a.union(d) == a)
+
+    @staticmethod
+    def _test_intersection(a, b, c, d):
+        intersect_ab = a.intersection(b)
+        assert_(np.array_equal(intersect_ab.ix, np.arange(30, 100)))
+        assert_(np.array_equal(a.intersection(b), b.intersection(a)))
+        assert_(len(a.intersection(d)) == 0)
+
+    @staticmethod
+    def _test_difference(a, b, c, d):
+        difference_ab = a.difference(b)
+        assert_(np.array_equal(difference_ab.ix, np.arange(10, 30)))
+
+        difference_ba = b.difference(a)
+        assert_(np.array_equal(difference_ba.ix, np.arange(100, 140)))
+
+        assert_(a.difference(d) == a)
+
+    @staticmethod
+    def _test_symmetric_difference(a, b, c, d):
+        symdiff_ab = a.symmetric_difference(b)
+        assert_(np.array_equal(symdiff_ab.ix,
+                                 np.array(list(range(10, 30))
+                                          + list(range(100, 140)))))
+        assert_(np.array_equal(a.symmetric_difference(b),
+                               b.symmetric_difference(a)))
+
+    @staticmethod
+    def resegment(u, res_per_seg):
+        n_atoms = len(u.atoms)
+        n_residues = len(u.residues)
+        attrs = [
+            attr
+            for attr in u._topology.attrs
+            if not isinstance(attr, Segids)
+        ]
+        limit = int(np.ceil(len(u.residues) / res_per_seg))
+        segids = np.arange(1131).repeat(res_per_seg)[:len(u.residues)]
+        segids = np.array(['s' + str(x) for x in segids], dtype=object)
+        segidx, (segids,) = change_squash((segids,), (segids,))
+        print(segidx)
+        n_segments = len(segids)
+        attrs.append(Segids(segids))
+        top = Topology(n_atoms, n_residues, n_segments,
+                       attrs=attrs,
+                       atom_resindex=u.atoms.resids,
+                       residue_segindex=segidx)
+        return top
+
+    @staticmethod
+    def make_groups(u, level):
+        a = getattr(u, level)[10:100]
```

Comment:
If you use the default `make_Universe()` system, you should be able to get away with using only 5 segments to test sets.  There's nothing about the set operations which **requires** more than 5 objects